sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,29 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n + 2
```

Comment:
Does `nterms = n` not suffice?

Reply:
yes I agree we can change this line to `nterms = n` and can change `cf != 0` to `cf > 0` then we dont have to apply `nterms < 0 ` check later.